### PR TITLE
[CDAP-16287] Added FLL for Azure plugins

### PR DESCRIPTION
**Pull request by vravish**

### CODE REVIEW

File: filesource-common/pom.xml [hunk 1]
```diff
@@ -52,6 +60,14 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
```
Comment (yaojiefeng) [inline filesource-common/pom.xml]: Why do we need this change?

File: adls-plugins/pom.xml [hunk 2]
```diff
@@ -75,6 +75,14 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
```
Comment (yaojiefeng) [inline adls-plugins/pom.xml]: why do we need this change

File: adls-plugins/src/main/java/io/cdap/plugin/sink/ADLSBatchSink.java [hunk 3]
```diff
@@ -115,6 +117,18 @@ public void prepareRun(BatchSinkContext context) throws Exception {
...
+
+    LineageRecorder lineageRecorder = new LineageRecorder(context, config.referenceName);
+    Schema schema = context.getInputSchema();
+    if (schema != null) {
+      lineageRecorder.createExternalDataset(schema);
+
+      if (schema.getFields() != null) {
+        lineageRecorder.recordWrite("Write", "Write to ADLS",
+            schema.getFields().stream().map(Schema.Field::getName).collect(
+                Collectors.toList()));
```
Comment (yaojiefeng) [inline adls-plugins/src/main/java/io/cdap/plugin/sink/ADLSBatchSink.java]: nit: indentation

File: filesource-common/src/main/java/io/cdap/plugin/common/AbstractFileBatchSource.java [hunk 4]
```diff
@@ -150,6 +149,18 @@ public void prepareRun(BatchSourceContext context) throws Exception {
...
 
+    LineageRecorder lineageRecorder = new LineageRecorder(context, config.referenceName);
+    Schema schema = context.getOutputSchema();
+    if (schema != null) {
+      lineageRecorder.createExternalDataset(schema);
+
+      if (schema.getFields() != null) {
+        lineageRecorder.recordRead("Read", "Read from source.",
+            schema.getFields().stream().map(Schema.Field::getName).collect(
+                Collectors.toList()));
```
Comment (yaojiefeng) [inline filesource-common/src/main/java/io/cdap/plugin/common/AbstractFileBatchSource.java]: nit: indentation

File: filesource-common/pom.xml (hunk 1 above)
Comment (vravish) [reply filesource-common/pom.xml]: This is to set the maven version to 1.8+

File: adls-plugins/pom.xml (hunk 2 above)
Comment (vravish) [reply adls-plugins/pom.xml]: This is to set the maven version to 1.8+

Comment (yaojiefeng) [reply adls-plugins/pom.xml]: You can just update this in the main pom.xml to 1.8

File: filesource-common/pom.xml (hunk 1 above)
Comment (yaojiefeng) [reply filesource-common/pom.xml]: You can just update this in the main pom.xml to 1.8

File: filesource-common/src/main/java/io/cdap/plugin/common/AbstractFileBatchSource.java (hunk 4 above)
Comment (yaojiefeng) [reply filesource-common/src/main/java/io/cdap/plugin/common/AbstractFileBatchSource.java]: Should have 2 spaces as indentation, you can put this in the same line.

File: adls-plugins/src/main/java/io/cdap/plugin/sink/ADLSBatchSink.java [hunk 5]
```diff
@@ -115,6 +117,18 @@ public void prepareRun(BatchSinkContext context) throws Exception {
...
+
+    LineageRecorder lineageRecorder = new LineageRecorder(context, config.referenceName);
+    Schema schema = context.getInputSchema();
+    if (schema != null) {
+      lineageRecorder.createExternalDataset(schema);
+
+      if (schema.getFields() != null) {
+        lineageRecorder.recordWrite("Write", "Write to ADLS",
+                                    schema.getFields().stream().map(Schema.Field::getName)
+                                        .collect(Collectors.toList()));
```
Comment (yaojiefeng) [inline adls-plugins/src/main/java/io/cdap/plugin/sink/ADLSBatchSink.java]: Should have 2 spaces as indentation

File: filesource-common/pom.xml (hunk 1 above)
Comment (vravish) [reply filesource-common/pom.xml]: Done

File: filesource-common/src/main/java/io/cdap/plugin/common/AbstractFileBatchSource.java (hunk 4 above)
Comment (vravish) [reply filesource-common/src/main/java/io/cdap/plugin/common/AbstractFileBatchSource.java]: Done

File: adls-plugins/pom.xml (hunk 2 above)
Comment (vravish) [reply adls-plugins/pom.xml]: Done

File: adls-plugins/src/main/java/io/cdap/plugin/sink/ADLSBatchSink.java (hunk 5 above)
Comment (vravish) [reply adls-plugins/src/main/java/io/cdap/plugin/sink/ADLSBatchSink.java]: Done

**Review by yaojiefeng — APPROVED** [review]

**Comment by yaojiefeng** [comment]
Please squash the commits before merge

**Pull request closed without merging**